Windows backslashes make the LSP server unusable

rcjsuen opened the pull request:
1. Have a `../git/tmp/src/Test.java` structure.
2. Launch the java-lsp Node server.

```
$ node modules/orionode/server.js ../git
```

3. Open Orion in your browser.
4. Open your `Test.java` file.
5. Make some changes so that the LSP server starts. Wait for the Java LSP initialization to complete.
6. Make changes again to force document change events to be sent.

```
DidChangeTextDocumentParams [
  textDocument = VersionedTextDocumentIdentifier [
    version = 1
    uri = "file:///C:/Users/Remy/code/eclipse/git\tmp\Test.java"
  ]
  uri = null
  contentChanges = ArrayList (
    TextDocumentContentChangeEvent [
      range = Range [
        start = Position [
          line = 2
          character = 2
        ]
        end = Position [
          line = 2
          character = 2
        ]
      ]
      rangeLength = 0
      text = " "
    ]
  )
]
```

7. An exception will occur.

```
java.net.URISyntaxException: Illegal character in path at index 38: file:///C:/Users/Remy/code/eclipse/git\tmp\src\Test.java
	at java.net.URI$Parser.fail(URI.java:2848)
	at java.net.URI$Parser.checkChars(URI.java:3021)
	at java.net.URI$Parser.parseHierarchical(URI.java:3105)
	at java.net.URI$Parser.parse(URI.java:3053)
	at java.net.URI.<init>(URI.java:588)
	at org.eclipse.jdt.ls.core.internal.JDTUtils.toURI(JDTUtils.java:560)
	at org.eclipse.jdt.ls.core.internal.JDTUtils.resolveCompilationUnit(JDTUtils.java:105)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler.handleSaved(DocumentLifeCycleHandler.java:214)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler.access$3(DocumentLifeCycleHandler.java:211)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler$4.run(DocumentLifeCycleHandler.java:101)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2262)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler.didSave(DocumentLifeCycleHandler.java:98)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.didSave(JDTLanguageServer.java:373)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:51)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:126)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:165)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:136)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:149)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:77)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:84)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

paulweb515 commented:
Looks good.  Could you please amend your commit and update the first line of your comment to look like the bug title, similar to https://wiki.eclipse.org/Orion/Contributing_Code#Commit_your_changes

paulweb515 commented:
That looks like you've changed the title of the PR.  I need you to amend the commit message and force-push your change.  Thanks.

rcjsuen committed:
Convert Windows backslashes to slashes to form a proper URI

When Orion server sends the URI of a file on Windows to the LSP
server, it will include a mixture of slashes and backslashes.
However, backslashes aren't allowed in a URI so we should convert
any backslashes to slashes before sending it to the server.

Signed-off-by: Remy Suen <remy.suen@gmail.com>
